jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/go/GoHelper.kt

```diff
@@ -45,51 +52,66 @@ fun inferSourceRoot(project: Project, virtualFile: VirtualFile): VirtualFile? {
     }
 }
 
-suspend fun createGoDebugProcess(
-    environment: ExecutionEnvironment,
-    state: SamRunningState,
-    debugHost: String,
-    debugPorts: List<Int>
-): XDebugProcessStarter {
-    val executionResult = state.execute(environment.executor, environment.runner)
+object GoDebugHelper : CoroutineScope by ApplicationThreadPoolScope("GoDebugHelper") {
+    // Reliable start message printed when delve starts
+    // Comes from: https://github.com/go-delve/delve/blob/f5d2e132bca763d222680815ace98601c2396517/service/debugger/debugger.go#L187
+    private const val startMessage = "launching process with args"
 
-    return object : XDebugProcessStarter() {
+    // TODO see https://youtrack.jetbrains.com/issue/GO-10775 for "Debugger disconnected unexpectedly" when the lambda finishes
+    suspend fun createGoDebugProcess(
+        debugHost: String,
+        debugPorts: List<Int>,
+        context: Context
+    ): XDebugProcessStarter = object : XDebugProcessStarter() {
         override fun start(session: XDebugSession): XDebugProcess {
-            val process = createDelveDebugProcess(session, executionResult)
+            val process = DlvDebugProcess(session, DlvRemoteVmConnection(DlvDisconnectOption.KILL), null, true)
```

Comment:
any ramifications due to a null ExecutionResult?

Reply:
Yep, it doesn't try to execute the state in a loop. it's marked `@nullable` so this is the correct usage